src/stores/events/EventsDataStore.ts

```diff
@@ -284,14 +292,34 @@ export default class EventsDataStore {
 					if (cachedRootNode) {
 						rootNode = cachedRootNode;
 					} else {
-						rootNode = getErrorEventTreeNode(rootNode?.parentId || parentId);
+						rootNode = unknownRoot;
+						if (currentParentId) {
+							this.parentChildrensMap.set(unknownRoot.eventId, [
+								...(this.parentChildrensMap.get(unknownRoot.eventId) || []),
+								...(this.parentChildrensMap.get(currentParentId) || []),
+							]);
+							this.parentChildrensMap.delete(currentParentId);
+							this.rootEventIds = [...this.rootEventIds, currentParentId];
+						}
 					}
 					parentNodes.unshift(rootNode);
 				}
 
 				parentNodes.forEach(eventNode => {
 					if (!this.eventsCache.has(eventNode.eventId)) {
-						this.eventsCache.set(eventNode.eventId, eventNode);
+						if (eventNode.eventId !== unknownRoot.eventId) {
+							this.eventsCache.set(eventNode.eventId, eventNode);
+						} else {
+							const eventsMap: Map<string, EventTreeNode> = observable.map(
+								new Map<string, EventTreeNode>([
+									...this.eventsCache,
+									// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+									[currentParentId!, unknownRoot],
```

Comment:
There should be only one `unknownRoot` which contains another unknown events, we might want to distinguish them. So maybe besides `unknownRoot` we should have `unknownEventNode`